### PR TITLE
feat: capitalize multiword documentation title and remove dashes

### PR DESCRIPTION
**Pull request by matthieusb**
[troll]Most groundbreaking feature of all time ![/troll]

A bit of a naive approach to it, but it seems to work.

### CODE REVIEW

File: src/cli/create-component.ts [hunk 1]
```diff
@@ -7,12 +7,18 @@ const createFile = (folderPath: string, filename: string): void => fse.ensureFil
...
-  const content = `## ${component}\n`;
+  const content = `## ${createDocumentationTitle(component)}\n`;
   const file = `${component}.md`;
   createFile(componentDirectory, file);
   fs.writeFileSync(path.resolve(componentDirectory, file), content);
 };
 
+const createDocumentationTitle = (component: string): string => {
+  const componentNameWithoutDash = component.replace(/-/g, ' ');
+
```
Comment (matthieusb) [inline src/cli/create-component.ts:18]: Pas très utile ce retour à la ligne

File: src/cli/create-component.ts [hunk 2]
```diff
@@ -7,12 +7,18 @@ const createFile = (folderPath: string, filename: string): void => fse.ensureFil
...
+  return componentNameWithoutDash.length > 1 ? componentNameWithoutDash.charAt(0).toUpperCase() + componentNameWithoutDash.slice(1) : componentNameWithoutDash.toUpperCase();
```
Comment (Gnuk) [inline src/cli/create-component.ts:19]: Maybe move to capitalize function

File: test/cli/create-component.spec.ts [hunk 3]
```diff
@@ -2,18 +2,18 @@ import * as fs from 'fs';
 import * as path from 'path';
 import * as rimraf from 'rimraf';
 import * as fse from 'fs-extra';
-import { createComponent } from '@/cli/create-component';
+import {createComponent} from '@/cli/create-component';
```
Comment (matthieusb) [inline test/cli/create-component.spec.ts:5]: Merci Intellij

Comment (matthieusb) [reply test/cli/create-component.spec.ts:5]: A supprimer j'imagine ? Ou c'est eslint qui fait des bêtises ?

File: src/cli/create-component.ts [hunk 4]
```diff
@@ -7,12 +7,18 @@ const createFile = (folderPath: string, filename: string): void => fse.ensureFil
...
 const createDocumentation = (componentDirectory: string, component: string): void => {
-  const content = `## ${component}\n`;
+  const content = `## ${createDocumentationTitle(component)}\n`;
   const file = `${component}.md`;
   createFile(componentDirectory, file);
   fs.writeFileSync(path.resolve(componentDirectory, file), content);
 };
 
+const createDocumentationTitle = (component: string): string => {
+  const componentNameWithoutDash = component.replace(/-/g, ' ');
```
Comment (Gnuk) [inline src/cli/create-component.ts:17]: maybe try `component.split('-').join(' ')` to not use a Regex ?

File: test/cli/create-component.spec.ts (hunk 3 above)
Comment (Gnuk) [reply test/cli/create-component.spec.ts:5]: Maybe eslint don't know

File: src/cli/create-component.ts (hunk 4 above)
Comment (matthieusb) [reply src/cli/create-component.ts:17]: I feel that you have a problem with regexes. I'll make the change.

File: src/cli/create-component.ts (hunk 2 above)
Comment (matthieusb) [reply src/cli/create-component.ts:19]: Done

File: src/cli/create-component.ts (hunk 4 above)
Comment (matthieusb) [reply src/cli/create-component.ts:17]: Done

**Review by Gnuk — APPROVED** [review]

**Pull request merged**